pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -131,14 +131,29 @@ func (esc *endpointSliceController) onEventInternal(_, ep *v1.EndpointSlice, eve
 			continue
 		}
 
-		esc.c.opts.XDSUpdater.ConfigUpdate(&model.PushRequest{
-			Full: features.EnableHeadlessService,
-			// TODO: extend and set service instance type, so no need to re-init push context
-			ConfigsUpdated: sets.New(model.ConfigKey{Kind: kind.ServiceEntry, Name: modelSvc.Hostname.String(), Namespace: svc.Namespace}),
-
-			Reason: model.NewReasonStats(model.HeadlessEndpointUpdate),
-		})
-		break
+		// pure HTTP headless services should not need a full push since they do not
+		// require a Listener based on IP: https://github.com/istio/istio/issues/48207
+		pureHTTP := true
+		for _, p := range modelSvc.Ports {
+			if !p.Protocol.IsHTTP() {
+				pureHTTP = false
+				break
+			}
+		}
+		// ConfigUpdate should be handled by pushEDS above
+		if !pureHTTP {
+			esc.c.opts.XDSUpdater.ConfigUpdate(&model.PushRequest{
+				Full: features.EnableHeadlessService,
```

Comment:
@GregHanson We donot need to run into ConfigUpdate if features.EnableHeadlessService = false and pureHttp, because pushEds has trigger non full push